src/pydvl/utils/parallel/backend.py

```diff
@@ -15,10 +17,89 @@
 
 T = TypeVar("T")
 
-_PARALLEL_BACKEND: Optional["RayParallelBackend"] = None
+_PARALLEL_BACKENDS: Dict[str, "BaseParallelBackend"] = {}
 
 
-class RayParallelBackend:
+class BaseParallelBackend(ABC):
+    """Abstract base class for all parallel backends"""
+
+    config: Dict[str, Any] = {}
+
+    @abstractmethod
+    def get(self, v: Any, *args, **kwargs):
+        ...
+
+    @abstractmethod
+    def put(self, v: Any, *args, **kwargs) -> Any:
+        ...
+
+    @abstractmethod
+    def wrap(self, *args, **kwargs) -> Any:
+        ...
+
+    @abstractmethod
+    def wait(self, v: Any, *args, **kwargs) -> Any:
+        ...
+
+    @abstractmethod
+    def effective_n_jobs(self, n_jobs: Optional[int]) -> int:
+        ...
+
+    def __repr__(self) -> str:
+        return f"<{self.__class__.__name__}: {self.config}>"
+
+
+class SequentialParallelBackend(BaseParallelBackend):
+    """Class used to run jobs sequentiall and locally. It shouldn't
+    be initialized directly. You should instead call `init_parallel_backend`.
+
+    :param config: instance of :class:`~pydvl.utils.config.ParallelConfig` with number of cpus
+
+    :Example:
+
+    >>> from pydvl.utils.parallel.backend import SequentialParallelBackend
+    >>> from pydvl.utils.config import ParallelConfig
+    >>> config = ParallelConfig(backend="sequential")
+    >>> parallel_backend = SequentialParallelBackend(config)
+    >>> parallel_backend
+    <SequentialParallelBackend: {'num_cpus': None}>
+
+    """
+
+    def __init__(self, config: ParallelConfig):
+        config_dict = asdict(config)
+        config_dict.pop("backend")
+        config_dict.pop("address")
+        config_dict["num_cpus"] = config_dict.pop("n_local_workers")
+        self.config = config_dict
+
+    def get(self, v: Any, *args, **kwargs):
+        return v
+
+    def put(self, v: Any, *args, **kwargs) -> Any:
```

Comment:
Shouldn't this store `v` in e.g. a `dict`, and return an id so that `get()` works on it? This would probably simplify things later (avoid handling different cases depending on the backend), right?

Reply:
This sequential class is just a dummy class that returns the same object. It's actually an error that put returns nothing here, I have fixed it locally and will push it when I'm done with the current round of fixing the type hints.